src/collection-view.js

```diff
@@ -553,12 +482,27 @@ var CollectionView = Backbone.View.extend({
     if (!view.supportsDestroyLifecycle) {
       triggerMethodOn(view, 'before:destroy', view);
     }
+
     // call 'destroy' or 'remove', depending on which is found
     if (view.destroy) {
       view.destroy();
     } else {
+      const shouldTriggerDetach = !!view._isAttached;
```

Comment:
I mentioned on region in a very similar spot that it might be nice to move the else logic into a method named something like `_destroyBackboneView` but now that there are 2 places we're doing this, I think a tiny module might be good so that we don't accidentally change the behavior in one place and not the other.